archinstall/lib/luks.py

```diff
@@ -97,120 +83,163 @@ def encrypt(self, partition :Partition,
 			'--hash', hash_type,
 			'--key-size', str(key_size),
 			'--iter-time', str(iter_time),
-			'--key-file', os.path.abspath(key_file),
+			'--key-file', str(key_file),
 			'--use-urandom',
-			'luksFormat', partition.path,
+			'luksFormat', str(self.luks_dev_path),
 		])
 
 		try:
 			# Retry formatting the volume because archinstall can some times be too quick
 			# which generates a "Device /dev/sdX does not exist or access denied." between
 			# setting up partitions and us trying to encrypt it.
+			cmd_handle = None
 			for i in range(storage['DISK_RETRY_ATTEMPTS']):
 				if (cmd_handle := SysCommand(cryptsetup_args)).exit_code != 0:
 					time.sleep(storage['DISK_TIMEOUTS'])
 				else:
 					break
 
-			if cmd_handle.exit_code != 0:
-				raise DiskError(f'Could not encrypt volume "{partition.path}": {b"".join(cmd_handle)}')
+			if cmd_handle is not None and cmd_handle.exit_code != 0:
+				output = str(b''.join(cmd_handle))
+				raise DiskError(f'Could not encrypt volume "{self.luks_dev_path}": {output}')
 		except SysCallError as err:
-			if err.exit_code == 1:
-				log(f'{partition} is being used, trying to unmount and crypt-close the device and running one more attempt at encrypting the device.', level=logging.DEBUG)
-				# Partition was in use, unmount it and try again
-				partition.unmount()
-
-				# Get crypt-information about the device by doing a reverse lookup starting with the partition path
-				# For instance: /dev/sda
-				SysCommand(f'bash -c "partprobe"')
-				devinfo = json.loads(b''.join(SysCommand(f"lsblk --fs -J {partition.path}")).decode('UTF-8'))['blockdevices'][0]
-
-				# For each child (sub-partition/sub-device)
-				if len(children := devinfo.get('children', [])):
-					for child in children:
-						# Unmount the child location
-						if child_mountpoint := child.get('mountpoint', None):
-							log(f'Unmounting {child_mountpoint}', level=logging.DEBUG)
-							SysCommand(f"umount -R {child_mountpoint}")
-
-						# And close it if possible.
-						log(f"Closing crypt device {child['name']}", level=logging.DEBUG)
-						SysCommand(f"cryptsetup close {child['name']}")
+			if err.exit_code == 256:
```

Comment:
```suggestion
			if err.exit_code == 1:
```

This is a regression from f611310 which looks to be merged in from `master`.

Reply:
Potentially not rebased yet, I'll have a look

Reply:
@codefiles I think the merge conflicts may have been resolved incorrectly, thanks for finding this! I keeps getting more difficult the longer the PR is open to keep it up to date with master :(